src/pages/Map.js

```diff
@@ -6,14 +6,16 @@ import LocationPin from '../components/LocationPin.js';
 import { LocationsContext } from '../context/LocationsContext';
 
 export default function Map() {
-  const [showWindow, setShowWindow] = useState({ show: false, id: '' });
-
-  const locationsContext = useContext(LocationsContext);
+  //refactored context
+  const {
+    locations,
+    zoomLevel,
+    coordinates,
+    recenter,
+    setRecenter,
+  } = useContext(LocationsContext);
 
-  //set locations, zoomLevel & coordinates based on context
-  const locations = locationsContext.locations;
-  const zoomLevel = locationsContext.zoomLevel;
-  const coordinates = locationsContext.coordinates;
+  const [showWindow, setShowWindow] = useState({ show: false, id: '' });
```

Comment:
Is `id: ''` here meant to be a falsey value? A bit more idiomatic JavaScript would be to set this to `null` or `undefined`

How I use those 2 values:

- `null` means I am deliberately setting it to a non-value
- `undefined` means the variable has been created, but has not yet been assigned any value deliberately

Reply:
I see! I think it could be undefined. The key is already created! We haven't set the value yet in state until we click the child component as assign it the respective ket.